src/game_objects.rs

```diff
@@ -44,80 +87,85 @@ pub struct Card {
     pub on_turn_end: Vec<Effect>,
     pub on_strike: Vec<Effect>,
     pub on_defend: Vec<Effect>,
-    pub cost: i16,
-    pub cost_currency: String,
+    pub cost: Cost,
     pub draw_to: DrawTo
 }
 
-pub type CardFactory = HashMap<String, Card>;
-
-#[derive(Serialize, Deserialize, Debug, Clone, Default)]
-pub struct Hand {
-    pub size: usize,
-    pub cards: Vec<Card>
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct CardContainer {
+    pub zone: BoardZone,
+    pub cards: Vec<Card>,
+    pub size: Option<usize>
 }
 
-impl Hand {
+impl CardContainer {
+    pub fn new(zone: BoardZone) -> Self {
+        Self { zone: zone, cards: Vec::<Card>::new(), size: None }
+    }
+
+    pub fn new_sized(zone: BoardZone, size: usize) -> Self {
+        Self { zone: zone, cards: Vec::<Card>::with_capacity(size), size: Some(size) }
+
+    }
+
+    pub fn add(&mut self, card: Card) {
+        self.cards.push(card)
+    }
+
+    pub fn empty(&self) -> bool {
+        self.cards.is_empty()
+    }
+
+    pub fn break_one(&mut self) {
```

Comment:
This is a weird method name: it makes sense only for Buildings. Suggestion: remove_first() (or just discard it and call remove(0)).

Reply:
Discarded, it is simple and used in one place only